test/acceptance/test/ui_tests.go

```diff
@@ -75,16 +80,142 @@ var _ = Describe("Weave GitOps UI Test", func() {
 		Expect(webDriver.Destroy()).To(Succeed())
 	})
 
-	It("SmokeTest - Verify gitops can run UI without apps installed", func() {
+	It("SmokeTest - Verify gitops can add apps from the UI to an empty cluster", func() {
+		var addAppPage *pages.AddAppPageElements
+		var repoAbsolutePath string
+		tip := generateTestInputs()
+		appName := tip.appRepoName
+		private := true
+		appRepoRemoteURL := "ssh://git@github.com/" + GITHUB_ORG + "/" + tip.appRepoName + ".git"
 
-		dashboardPage := pages.Dashboard(webDriver)
-		expectedTitle := "Weave GitOps"
+		dashboardPage = pages.GetDashboardPageElements(webDriver)
+		addAppPage = pages.GetAddAppPageElements(webDriver)
 
-		By("Then I should be able to navigate to WeGO dashboard", func() {
-			Expect(webDriver.Navigate(WEGO_UI_URL)).To(Succeed())
-			str, _ := webDriver.Title()
-			Eventually(str, THIRTY_SECOND_TIMEOUT).Should(ContainSubstring(expectedTitle))
-			Expect(dashboardPage.ApplicationTab).Should(BeFound())
+		defer deleteRepo(tip.appRepoName, gitproviders.GitProviderGitHub, GITHUB_ORG)
+		defer deleteWorkload(tip.workloadName, tip.workloadNamespace)
+
+		By("When I navigate to Add Application page", func() {
+			Expect(dashboardPage.AddAppButton.Click()).To(Succeed())
+		})
+
+		By("Then I should see Add Applcation page", func() {
+			Expect(addAppPage.AddAppHeader.Text()).Should(ContainSubstring(addApplicationPageHeader))
 		})
+
+		By("When I create an app repo with workload that does not already exist", func() {
+			deleteRepo(tip.appRepoName, gitproviders.GitProviderGitHub, GITHUB_ORG)
+			repoAbsolutePath = initAndCreateEmptyRepo(tip.appRepoName, gitproviders.GitProviderGitHub, private, GITHUB_ORG)
+			gitAddCommitPush(repoAbsolutePath, tip.appManifestFilePath)
+		})
+
+		By("And I install gitops to my active cluster", func() {
+			installAndVerifyWego(WEGO_DEFAULT_NAMESPACE, appRepoRemoteURL)
+		})
+
+		By("And I add application details in Application form", func() {
+			_ = addAppPage.AppName.Fill(appName)
+			_ = addAppPage.AppRepoURL.Fill(appRepoRemoteURL)
+			_ = addAppPage.PathToManifests.Fill("/")
+		})
+
+		By("And default form values are present", func() {
+			Expect(addAppPage.AppNamespace.Attribute("value")).Should(ContainSubstring(WEGO_DEFAULT_NAMESPACE))
+			Expect(addAppPage.Branch.Attribute("value")).Should(ContainSubstring("main"))
+		})
+
+		By("And auto-merge is turned on", func() {
+			_ = addAppPage.AutoMergeCheck.Check()
+		})
+
+		By("And I submit the Add App form", func() {
+			Expect(addAppPage.SubmitButton.Click()).To(Succeed())
+		})
+
+		By("Then I should see Authentication Button pop up", func() {
+			_ = Expect(addAppPage.AuthenticationButton.Visible()).To(BeTrue())
```

Comment:
What does `.Visible()` check for? The presence of the element in the DOM?

Reply:
No, if the element is visible to the user. As soon as you hit `Add Application` Button, the `Github Authentication` button pops up. The point of this function is to verify the button actually pops up in the UI

Reply:
Visibility can take lots of forms. I can change it with css via `display: none`, `visibility: none`, `opacity: 0`, or remove it from the DOM entirely.

Reply:
I think the use of visible matcher can make condition more understandable.
`Expect(addAppPage.AuthenticationButton).Should(BeVisible())`

The matcher passes if the selection refers to element is displayed on the page.
Incase the element is not present in the DOM the webdriver throws an error straightaway.

Reply:
I'm removing this for now since it requires a whole workaround.